test/resources/object-transforms.js

```diff
@@ -0,0 +1,692 @@
+                    fields: ["sku", "id"],
+                    to: "moved"
+                }]
+            }
+        }
+    },
+    transformedRequest: {
+        searchProducts: {
+            results: {
+                moved: {
+                    id: {},
+                    sku: {}
+                },
+                ghost: {
+                    name: {}
+                }
+            }
+        }
+    }
+};
+
+const deltesEmptyFieldAfterMove = {
+    initialRequest: {
+        searchProducts: {
+            results: {
+                id: {},
+                sku: {}
+            }
+        }
+    },
+    transformRules: {
+        searchProducts: {
+            moveFields: [{
+                from: "results",
+                fields: ["sku", "id"],
+                to: "moved"
+            }]
+        }
+    },
+    transformedRequest: {
+        searchProducts: {
+            moved: {
+                id: {},
+                sku: {}
+            },
+        }
+    }
+};
+
+const deltesConditionalEmptyFieldAfterMove = {
+    initialRequest: {
+        searchProducts: {
+            results: {
+                product: {
+                    fields: {
+                        id: {},
+                        sku: {}
+                    }
+                }
+            }
+        }
+    },
+    transformRules: {
+        searchProducts: {
+            moveFields: [{
+                from: "results.product.fields",
+                fields: ["sku", "id"],
+                to: "moved"
+            }]
+        }
+    },
+    transformedRequest: {
+        searchProducts: {
+            moved: {
+                id: {},
+                sku: {}
+            }
+        }
+    }
+};
+
+const addsArgs = {
+    initialRequest: {
+        searchProducts: {
+            results: {
+                id: {},
+                sku: {}
+            }
+        }
+    },
+    transformRules: {
+        searchProducts: {
+            args: {
+                text: "shorts"
+            }
+        }
+    },
+    transformedRequest: {
+        searchProducts: {
+            __args: {
+                text: "shorts"
+            },
+            results: {
+                id: {},
+                sku: {}
+            }
+        }
+    }
+};
+
+const mergeArgs = {
+    initialRequest: {
+        searchProducts: {
+            __args: {
+                text: "meskwielt"
+            },
+            results: {
+                id: {},
+                sku: {}
+            }
+        }
+    },
+    transformRules: {
+        searchProducts: {
+            args: {
+                limit: 20
+            }
+        }
+    },
+    transformedRequest: {
+        searchProducts: {
+            __args: {
+                text: "meskwielt",
+                limit: 20
+            },
+            results: {
+                id: {},
+                sku: {}
+            }
+        }
+    }
+};
+
+const removesEmptyObjectAfterIgnore = {
```

Comment:
looks like a very similar test than `deleteEmptyObject`, do we really need this extra one?